buffer/src/main/java/io/netty5/buffer/api/bytebuffer/NioBuffer.java

```diff
@@ -1183,7 +1183,7 @@ private void handleWriteAccessBoundsFailure(int index, int size, boolean mayExpa
             throw bufferIsReadOnly(this);
         }
         int capacity = capacity();
-        if (mayExpand && index >= 0 && index <= capacity && woff + size <= implicitCapacityLimit && isOwned()) {
+        if (mayExpand && index >= 0 && index <= capacity && index + size <= implicitCapacityLimit && isOwned()) {
```

Comment:
Are you sure this change is correct ? /cc @chrisvest

Reply:
For all cases where `mayExpand` is `true`, the `index` should always be equal to the write offset, so they should be equivalent.
I think it's a bit clearer what the check is about when we use write offset, though.

Reply:
If it is the case, I think it would be clearer if we split it into two functions, one with single argument 'size'(current mayExpand = true, which will checks using woff), and the other one with 'index' + 'size'(current mayExpand = false).

Reply:
I reverted it. Could you confirm it?